traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -846,6 +875,15 @@ func getServers(h http.Header, params map[string]string, tx *sqlx.Tx, user *auth
 	}
 
 	where, orderBy, pagination, queryValues, errs := dbhelpers.BuildWhereAndOrderByAndPagination(params, queryParamsToSQLCols)
+
+	// get the origin server count
+	if _, ok := params["dsId"]; ok {
+		serverCountOrigin, err = getServerCount(tx, serverCountQuery+originServerQuery, queryValues)
+		if err != nil {
+			return nil, 0, nil, fmt.Errorf("failed to get origin servers count: %v", err), http.StatusInternalServerError, nil
+		}
+	}
```

Comment:
Why is it necessary to count origin servers separately? We should avoid adding additional queries when we can.

Reply:
I've made the change so that we are accounting for the origin servers in the same query.